### PR TITLE
approval flow loading text

### PR DESCRIPTION
**Pull request by bergarces**
## Description

Adds a method to set loading text for the approval flow pages.

## Changes

- **ADDED**: Adds method `setFlowLoadingText` to the controller

## References

Fixes https://github.com/MetaMask/MetaMask-planning/issues/567

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [X] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [X] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/approval-controller/src/ApprovalController.ts
Comment (bergarces) [inline packages/approval-controller/src/ApprovalController.ts:721]: I'm equally ok with either this or having `loadingText` as an optional parameter which sets it to null.

File: packages/approval-controller/src/ApprovalController.ts
Comment (bergarces) [inline packages/approval-controller/src/ApprovalController.ts:1]: Anyone is of the opinion that `endFlow` should set the loading text back to `null`? It might be inconvenient in the event of an approval flow within another approval flow, but I think that's going to be a very rare occurrence.

On the other hand, not clearing it after a previous approval flow could cause some interesting bugs depending in which order approval flows happen.

An alternative to this would be to add a `loadingText` parameter when a flow starts and set it back to default if nothing is passed.

And even though the ticket explicitly says not to, another alternative would be for the loading text to be linked to an approval flow.

File: packages/approval-controller/src/ApprovalController.ts
Comment (matthewwalsh0) [reply packages/approval-controller/src/ApprovalController.ts:721]: I vote this, feels like overkill to have options here, at least for now, plus more explicit to provide null instead of an empty object.

File: packages/approval-controller/src/ApprovalController.ts
Comment (bergarces) [reply packages/approval-controller/src/ApprovalController.ts:1]: So...

Without doing text by `flowId`, we can't just clear the text at `endFlow`, as that would inadvertently clear the parent loading text.

If we attach the flow to a specific `flowId`, then it won't matter. We can even come up with ways of inheriting the parent text if one is not provided.

Having said that, it's going to be difficult to predict what the functionality should exactly be if we don't have any use case for a nested approval flow.

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**
